modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DataNodesManager.java

```diff
@@ -1013,7 +1048,7 @@ private CompletableFuture<Void> msInvokeWithRetry(
                         return null;
                     }
                 })
-                .thenCompose(c -> nullCompletedFuture());
+                .thenApply(c -> null);
```

Comment:
I still don't understand this code, why is this line needed? What's going to happen to the future from this line:

```
return msInvokeWithRetry(iifSupplier, attemptsLeft - 1, zone);
```